src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,113 @@
+import { useParams, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../types';
+import { PersonLink } from '../types/PersonLink';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[];
+};
+
+function getVisiblePeople(
+  people: Person[],
+  sex: string,
+  query: string,
+  centuries: string[],
+  sort: string,
+  order: string,
+) {
+  let copyPeople = [...people];
+
+  if (sex) {
+    copyPeople = copyPeople.filter(person => person.sex === sex);
+  }
+
+  if (query) {
+    copyPeople = copyPeople.filter(
+      person =>
+        person.name.toLowerCase().includes(query) ||
+        person.motherName?.toLowerCase().includes(query) ||
+        person.fatherName?.toLowerCase().includes(query),
+    );
+  }
+
+  if (centuries.length !== 0) {
+    copyPeople = copyPeople.filter(person =>
+      centuries.includes(`${person.born + 100}`.slice(0, 2)),
+    );
+  }
+
+  if (sort) {
+    switch (sort) {
+      case 'name':
+        copyPeople.sort((el1, el2) => el1.name.localeCompare(el2.name));
+        break;
+      case 'sex':
+        copyPeople.sort((el1, el2) => el1.sex.localeCompare(el2.sex));
+        break;
+      case 'born':
+        copyPeople.sort((el1, el2) => el1.born - el2.born);
+        break;
+      case 'died':
+        copyPeople.sort((el1, el2) => el1.died - el2.died);
+        break;
+    }
+  }
+
+  if (order) {
+    return copyPeople.reverse();
+  }
+
+  return copyPeople;
+}
+
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slugId } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const findPersonByName = (name: string) => {
+    return people.find(person => person.name === name);
+  };
+
+  const peopleWithParents = people.map(person => {
+    const mother = person.motherName
+      ? findPersonByName(person.motherName)
+      : undefined;
+    const father = person.fatherName
+      ? findPersonByName(person.fatherName)
+      : undefined;
+
+    return { ...person, mother, father };
+  });
+
+  function sortClick(sortField: string) {
+    if (!order && !sort) {
+      return { sort: sortField, order: null };
```

Comment:
The `order` value should be 'asc' or 'desc' to indicate the sorting order. Returning `null` for `order` might not be the intended behavior. Consider using 'asc' or 'desc' instead.